packages/waku/src/lib/plugins/vite-plugin-deploy-deno.ts

```diff
@@ -18,22 +18,29 @@ import { serverEngine } from 'waku/unstable_hono';
 const distDir = '${distDir}';
 const publicDir = '${distPublic}';
 const loadEntries = () => import('${srcEntriesFile}');
+const configPromise = loadEntries().then((entries) => entries.loadConfig());
 const env = Deno.env.toObject();
 
-const app = new Hono();
-app.use(serveStatic({ root: distDir + '/' + publicDir }));
-app.use(serverEngine({ cmd: 'start', loadEntries, env }));
-app.notFound(async (c) => {
-  const file = distDir + '/' + publicDir + '/404.html';
-  try {
-    const info = await Deno.stat(file);
-    if (info.isFile) {
-      c.header('Content-Type', 'text/html; charset=utf-8');
-      return c.body(await Deno.readFile(file), 404);
-    }
-  } catch {}
-  return c.text('404 Not Found', 404);
-});
+const createApp = (app) => {
+  app.use(serveStatic({ root: distDir + '/' + publicDir }));
+  app.use(serverEngine({ cmd: 'start', loadEntries, env }));
+  app.notFound(async (c) => {
+    const file = distDir + '/' + publicDir + '/404.html';
+    try {
+      const info = await Deno.stat(file);
+      if (info.isFile) {
+        c.header('Content-Type', 'text/html; charset=utf-8');
+        return c.body(await Deno.readFile(file), 404);
+      }
+    } catch {}
+    return c.text('404 Not Found', 404);
+  });
+  return app;
+};
+
+const honoEnhancer =
+  (await configPromise).unstable_honoEnhancer || ((createApp) => createApp);
+const app = honoEnhancer(createApp)(new Hono());
```

Comment:
Maybe we can do the same for `vite-plugin-deploy-cloudflare.ts` using top-level await.